### ISSUE TITLE
Using a type alias with arguments in `case` block in a `match` statement gets the wrong error message with `parameters` instead of `arguments`

### ISSUE THREAD

**Issue by hyperkai** [python/mypy#20495]
*Memo:
- mypy test.py
- mypy 1.19.1
- Python 3.14
- Windows 11

Using a type alias with arguments in `case` block in a [match statement](https://docs.python.org/3/reference/compound_stmts.html#the-match-statement) gets the wrong error message with `parameters` instead of `arguments` as shown below:

```python
class Cls:
    def __init__(self, x: int, y: int):
        self.x = x
        self.y = y

type TA = Cls

match Cls(x=10, y=20):
    case TA(x=10, y=20):
        print("OK")
```

> error: Class pattern class must not be a type alias with type parameters

So, the error message should be with `arguments` instead of `parameters` as shown below:

> error: Class pattern class must not be a type alias with type arguments

**Comment by hyperkai** [comment]
I noticed that the error message itself below is wrong because `TA(x=10, y=20)` is a **caller** to instantiate `TA` but not **type arguments** in `case` block as shown below:

> error: Class pattern class must not be a type alias with type arguments

```python
class Cls:
    def __init__(self, x: int, y: int):
        self.x = x
        self.y = y

type TA = Cls

match Cls(x=10, y=20):
    case TA(x=10, y=20): # It's a caller to instantiate `TA` but not type arguments.
        print("OK")
```

So, the error message should be `error: "TypeAliasType" not callable` whose error message is already used when calling a type alias to instantiate it as shown below:

```python
class Cls:
    def __init__(self, x: int, y: int):
        self.x = x
        self.y = y

type TA = Cls

TA(x=10, y=20) # Error
```

> error: "TypeAliasType" not callable

In addition, using a generic type alias with type arguments gets `error: Invalid syntax` as shown below:

```python
class Cls:
    def __init__(self, x: int, y: int):
        self.x = x
        self.y = y

type TA = Cls
# type TA[T] = Cls

match Cls(x=10, y=20):
    case TA[int]: # Error
        print("OK")
```

> error: Invalid syntax

**Comment by A5rocks** [comment]
To be clear, the problematic behavior here is that mypy is saying that this type alias has type parameters when it clearly doesn't.
